packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -139,16 +140,53 @@ def _get_tx_data(
             )
             return tx_data
 
+        # Check that we have a RAW_TRANSACTION response
         if message.performative != ContractApiMessage.Performative.RAW_TRANSACTION:
             self.context.logger.warning(
                 f"get_raw_safe_transaction unsuccessful! Received: {message}"
             )
             return tx_data
 
+        # Send transaction
         tx_digest, rpc_status = yield from self.send_raw_transaction(
             message.raw_transaction
         )
 
+        # Guess gas strategy
+        gas_strategy = None
+        if (
+            "maxFeePerGas" in message.raw_transaction.body
+            and "maxPriorityFeePerGas" in message.raw_transaction.body
+        ):
+            gas_strategy = "eip"
+        if "gasPrice" in message.raw_transaction.body:
+            gas_strategy = "gas_station"
+
+        self.context.logger.info(f"Detected gas strategy: {gas_strategy}")
```

Comment:
Let's also open an issue for this part, so that we remove it when we can access the connection's configuration.

Reply:
#195

Reply:
Please note: even if we address that issue, we could only access to the default price strategy config, but an agent could be not using the default strategy (try_get_gas_pricing can be called with a different gas_price_strategy) so I think we still need to keep using this method.

Reply:
Yes, but neither our contract nor tx settlement skill will allow that behaviour. Also, if a user implements their own skill and does that, then it is a mistake. The strategy should always be set depending on the global configuration IMO.